packages/phasor/src/elements/surface-element.ts

```diff
@@ -4,15 +4,21 @@ import type * as Y from 'yjs';
 import type { IVec } from '../index.js';
 import type { Renderer } from '../renderer.js';
 import type { SurfaceManager } from '../surface.js';
-import { isPointIn } from '../utils/math-utils.js';
-import { deserializeXYWH, type SerializedXYWH } from '../utils/xywh.js';
+import { decomposeMatrix, isPointIn } from '../utils/math-utils.js';
+import {
+  deserializeXYWH,
+  type SerializedXYWH,
+  type XYWH,
+} from '../utils/xywh.js';
 
 export interface ISurfaceElement {
   id: string;
   type: string;
   xywh: SerializedXYWH;
   index: string;
   seed: number;
+
+  matrix: number[];
```

Comment:
I have concern about using matrix as source of truth, which may be overly powerful and will make it harder for future graphics optimizations. Sorry for requesting change but this could have been discussion before.

Reply:
Ok, bring `rotate`, `flipX`, `flipY` back.

Reply:
Ok, bring `rotate`, `flipX`, `flipY` back.